packages/react-server-dom-parcel/src/shared/ReactFlightClientConfigBundlerParcel.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {Thenable} from 'shared/ReactTypes';
+
+import type {ImportMetadata} from './ReactFlightImportMetadata';
+
+import {ID, NAME, BUNDLES} from './ReactFlightImportMetadata';
+
+export type ServerManifest = {
+  [string]: Array<string>,
+};
+export type SSRModuleMap = null;
+export type ModuleLoading = null;
+export type ServerConsumerModuleMap = null;
+export type ServerReferenceId = string;
+
+export opaque type ClientReferenceMetadata = ImportMetadata;
+
+// eslint-disable-next-line no-unused-vars
+export opaque type ClientReference<T> = {
+  // Module id.
+  id: string,
+  // Export name.
+  name: string,
+  // List of bundle URLs, relative to the distDir.
+  bundles: Array<string>,
+};
+
+export function prepareDestinationForModule(
+  moduleLoading: ModuleLoading,
+  nonce: ?string,
+  metadata: ClientReferenceMetadata,
+) {
+  return;
```

Comment:
By not implementing this, the thing you lose out on is the ability to emit early loading of the modules during SSR. So it would have to wait until hydration to kick off loading the bundles that will later be needed.

It's a little convoluted the way that's wired up in the other packages to only do that for SSR and not in the browser.

But the key is that you just need some way to provide the absolute URL that the browser would load.

Reply:
The `<Resources>` component we have inserts scripts and stylesheets needed by the page based on dependencies in the module graph, so I thought this wasn't needed at first. But thinking about it more, in theory there could be bundles that are discovered conditionally during render (e.g. if a component does a dynamic `import()`). I can add something similar to what the other packages are doing.

Looks like this only supports preloading JS and not CSS though? So in the case where resources were discovered during render, we'd preload the JS but the CSS would only be discovered during hydration. Or am I missing something?

Reply:
Actually I think we need to find a way to render a `<link>` element in the tree during render for dynamic imports. Otherwise there will be FOUC before hydration. This can be done manually by rendering `<Resources>` inside each dynamically imported component. I guess that would solve the issue for JS too. Would be cool if this could happen automatically though....

Reply:
@sebmarkbage I think I could potentially generate a proxy module when you use dynamic `import()` inside a server component (either for another server component or a client component), and emit wrappers that would insert the necessary `<link>` elements.

I think this can't be done in `prepareDestinationForModule` since that only knows about client components, and not about css needed for server components (when there are no client components). And it should suspend on the css to avoid FOUC (I think preinit does not?).

The hard part is knowing what is a valid component that should be wrapped by the proxy. We wouldn't want to unintentionally change the return type of a non-component function by wrapping it with extra JSX elements.

Should there be a way of annotating server component exports with their resources so that stylesheets can be inserted automatically? Manually adding `<Resources>` in every component that you dynamic `import()` might be hard to remember.